src/layout-grid.scss

```diff
@@ -2,44 +2,156 @@
 @import "./mixins";
 
 /*!
-.fd-layout-grid
+.fd-col
 */
 
-$block: #{$fd-namespace}-layout-grid;
+$block: #{$fd-namespace}-col;
+$blockContainer: #{$fd-namespace}-container;
+$blockRow: #{$fd-namespace}-row;
 
-.#{$block} {
-  @include fd-reset();
+@media (min-width: 0) {
+  .#{$blockContainer} {
+    @include fd-reset();
 
-  display: grid;
-  grid-template-columns: repeat(3, 1fr);
-  grid-gap: var(--fd-width-gutter, $fd-width--gutter);
+    display: flex;
+    flex-wrap: wrap;
+    flex: 0 0 100%;
+    position: relative;
+    margin: 0 -0.25rem;
 
-  &--no-gap {
-    grid-gap: 0;
+    &--no-gap {
+      margin: 0;
+    }
+
+    :last-child {
+      margin-bottom: 0;
+    }
+
+    .#{$blockRow} {
+      margin-left: 0;
+      margin-right: 0;
+    }
+  }
+
+  .#{$blockRow} {
+    @include fd-reset();
+
+    display: flex;
+    flex-wrap: wrap;
+    flex: 0 0 100%;
+    position: relative;
+    margin: 0 -0.25rem 0.5rem -0.25rem;
+
+    :last-child {
+      margin-bottom: 0;
+    }
+
+    &--inner-flex {
+      display: flex;
+    }
+
+    .#{$block} {
+      .#{$blockRow} + .#{$blockRow} {
+        margin-top: 0.5rem;
+      }
+    }
+  }
+
+  .#{$block} {
+    @include fd-reset();
+
+    padding: 0 0.25rem;
+    min-width: 100%;
+    max-width: 100%;
+
+    @for $n from 1 through 12 {
+      &--#{$n} {
+        min-width: ( ( $n / 12 ) *100%);
+        max-width: ( ( $n / 12 ) *100%);
+      }
+    }
+
+    @for $n from 1 through 12 {
+      &--offset-#{$n} {
+        margin-left: ( ( $n / 12 ) *100%);
+      }
+    }
 
-    & > * {
-      box-shadow: none;
+    &--no-gap {
+      padding: 0;
+    }
+
+    .#{$blockRow} {
+      margin: -0.25rem 0;
+    }
+  }
+}
+
+@media (min-width: 600px) {
+  .#{$blockContainer} {
+    margin: 0 -0.5rem;
+
+    .#{$blockRow} {
+      margin-left: 0;
+      margin-right: 0;
+    }
+
+    &--no-gap {
+      margin: 0;
+    }
+
+    :last-child {
+      margin-bottom: 0;
+    }
+  }
+
+  .#{$blockRow} {
+    margin-bottom: 1rem;
+
+    .#{$block} {
+      .#{$blockRow} + .#{$blockRow} {
+        margin-top: 1rem;
+      }
     }
   }
```

Comment:
This is overcomplicated, I would go for 
```scss
.#{$blockRow} {
    margin-bottom: 1rem;
    &:last-chlid {
        margin-bottom: 0;
    }
}
```

And if you set margin for sides, use `margin-left` and `margin-right`

Reply:
We cannot use the solution provided because the first child from a nested row cannot have a margin bottom. And yes i tried using first child selector, that solution does not work either. It is to specifically target a row with has older/higherDOM siblings